src/ndpp_ops.F90

```diff
@@ -0,0 +1,1083 @@
+module ndpp_ops
+
+  use ace_header,   only: XsListing
+  use constants
+  use dict_header,  only: DictCharInt, DICT_NULL
+  use error,        only: fatal_error
+  use global
+  use list_header,  only: ListElemInt
+  use math,         only: calc_rn
+  use ndpp_header,  only: GrpTransfer, Ndpp
+  use search
+  use string,       only: ends_with, to_lower, starts_with, to_str
+  use tally_header, only: TallyResult
+  use xml_interface
+
+  implicit none
+
+  contains
+
+!===============================================================================
+! NDPP_READ reads information from a pre-processed nuclear data file
+! as generated by the NDPP program and initializes the Ndpp object accordingly.
+!===============================================================================
+
+  subroutine ndpp_read(this, listing, get_scatt, get_nuscatt, get_chi_t, get_chi_p, &
+                       get_chi_d, scatt_order, is_nuc)
+    type(Ndpp), intent(inout) :: this ! Ndpp object to initialize
+    type(XsListing),  pointer, intent(in) :: listing ! Current NDPP data file
+    logical, intent(in) :: get_scatt   ! Whether or not to get scatt data
+    logical, intent(in) :: get_nuscatt ! Whether or not to get nuscatt data
+    logical, intent(in) :: get_chi_t   ! Whether or not to get total chi data
+    logical, intent(in) :: get_chi_p   ! Whether or not to get prompt chi data
+    logical, intent(in) :: get_chi_d   ! Whether or not to get delayed chi data
+    integer, intent(in) :: scatt_order ! Number of moments requested in tallies
+    logical, intent(in) :: is_nuc      ! Is our data a nuc or an sab?
+
+    integer       :: in = 7        ! file unit
```

Comment:
I see that you copied that from ace.F90.  I'd appreciate it if you replaced the `in` variable in ace.F90 as well.  I guess they should both be using a parameter variable from constants.F90 named `DATA_FILE_HANDLE` or something like that...